src/smc-webapp/project_actions.ts

```diff
@@ -2647,6 +2649,17 @@ export class ProjectActions extends Actions<ProjectStoreState> {
   close_free_warning(): void {
     this.setState({ free_warning_closed: true });
   }
+
+  async set_compute_image(new_image: string): Promise<void> {
```

Comment:
Harald -- this one little function *replaces most of the code* you added to smc-util/client.coffee, message.coffee, smc-hub/client.cofee, postgres-server-queries.coffee.   I spent a huge amount of time figuring out how to do this sensibly, rather than having to edit 5 files to do anything.

Reply:
looks nice, what I don't understand is how this checks that the image name is valid.